frontend/src/components/StateCircle/StateCircle.js

```diff
@@ -8,7 +9,19 @@ const StateCircle = ({ id, name, isFinal, cx, cy, selected, ...props }) => {
   // TODO: use prefix preference
   const displayName = name || `q${id}`
 
-  return <g transform={`translate(${cx}, ${cy})`} {...props}>
+  // TODO: use Callback
+  const handleStateMouseUp = e =>
+    dispatchEvent('state:mouseup', {
+      originalEvent: e,
+      state: { id, name, cx, cy },
+    })
+  const handleStateMouseDown = e =>
+    dispatchEvent('state:mousedown', {
+      originalEvent: e,
+      state: { id, name, cx, cy },
+    })
+
+  return <g transform={`translate(${cx}, ${cy})`} onMouseDown={handleStateMouseDown} onMouseUp={handleStateMouseUp} {...props}>
     {/* Yellow Circle */}
```

Comment:
Does the colour need to be mentioned here? It'll change for PDAs and Turing Machines so it won't work as a comment then